src/gui/virtualstudio.cpp

```diff
@@ -866,9 +898,17 @@ void VirtualStudio::getServerList(bool firstLoad, int index)
                 VsServerInfo* serverInfo = new VsServerInfo(this);
                 serverInfo->setIsManageable(
                     servers.at(i)[QStringLiteral("admin")].toBool());
-                QString status = servers.at(i)[QStringLiteral("status")].toString();
-                // Only include active servers, or servers that we can manage.
-                if (status == QLatin1String("Ready") || serverInfo->isManageable()) {
+                QString status     = servers.at(i)[QStringLiteral("status")].toString();
+                bool activeStudio  = status == QLatin1String("Ready");
+                bool managedStudio = servers.at(i)[QStringLiteral("managed")].toBool();
+                // Only iterate through servers that we want to show
+                if (!m_showSelfHosted && !managedStudio) {
+                    continue;
+                }
+                if (!m_showInactive && !activeStudio) {
+                    continue;
+                }
+                if (m_showInactive || activeStudio || serverInfo->isManageable()) {
```

Comment:
The behavior I expect is `if ( activeStudio || (serverInfo->isManageable() && m_showInactive))`. Extra parentheses for emphasis. I think we only want to show inactive studios you are the admin of? That's how it was before the filters.

Reply:
Done